tests/cases/compiler/bigintWithLib.ts

```diff
@@ -19,10 +19,18 @@ stringVal = bigintVal.toLocaleString('de-DE', { style: 'currency', currency: 'EU
 let bigIntArray: BigInt64Array = new BigInt64Array();
 bigIntArray = new BigInt64Array(10);
 bigIntArray = new BigInt64Array([1n, 2n, 3n]);
+bigIntArray = new BigInt64Array({0: 1n, 1: 2n, 2: 3n, length: 3});
 bigIntArray = new BigInt64Array([1, 2, 3]); // should error
+bigIntArray = new BigInt64Array({0: 1, 1: 2, 2: 3, length: 3}); // should error
 bigIntArray = new BigInt64Array(new ArrayBuffer(80));
 bigIntArray = new BigInt64Array(new ArrayBuffer(80), 8);
 bigIntArray = new BigInt64Array(new ArrayBuffer(80), 8, 3);
+bigIntArray = BigInt64Array.from([1n, 2n, 3n]);
+bigIntArray = BigInt64Array.from([1n, 2n, 3n], n => n * 10n);
+bigIntArray = BigInt64Array.from({0: 1n, 1: 2n, 2: 3n, length: 3});
+bigIntArray = BigInt64Array.from({0: 1n, 1: 2n, 2: 3n, length: 3}, n => n * 10n);

```

Comment:
re. [this conversation](https://github.com/microsoft/TypeScript/pull/58188#discussion_r1640355261), this is the case that necessitates keeping `ArrayLike` alongside `Iterable`.